tests/staggered_invert_test.cpp

```diff
@@ -126,243 +116,155 @@ void constructSpinorField(Float *res) {
   // since the input fields are unitarized.
-  gaugeParam->tadpole_coeff = 1.0;
+  gauge_param.tadpole_coeff = 1.0;
+
   if (dslash_type == QUDA_ASQTAD_DSLASH) {
-    gaugeParam->scale = -1.0/24.0;
-    if (eps_naik != 0) {
-      gaugeParam->scale *= (1.0+eps_naik);
-    }
+    gauge_param.scale = -1.0 / 24.0;
+    if (eps_naik != 0) { gauge_param.scale *= (1.0 + eps_naik); }
   } else {
-    gaugeParam->scale = 1.0;
+    gauge_param.scale = 1.0;
   }
-  gaugeParam->gauge_order = QUDA_MILC_GAUGE_ORDER;
-  gaugeParam->t_boundary = QUDA_ANTI_PERIODIC_T;
-  gaugeParam->staggered_phase_type = QUDA_STAGGERED_PHASE_MILC;
-  gaugeParam->gauge_fix = QUDA_GAUGE_FIXED_NO;
-  gaugeParam->type = QUDA_WILSON_LINKS;
+  gauge_param.gauge_order = QUDA_MILC_GAUGE_ORDER;
+  gauge_param.t_boundary = QUDA_ANTI_PERIODIC_T;
+  gauge_param.staggered_phase_type = QUDA_STAGGERED_PHASE_MILC;
+  gauge_param.gauge_fix = QUDA_GAUGE_FIXED_NO;
+  gauge_param.type = QUDA_WILSON_LINKS;
 
-  gaugeParam->ga_pad = 0;
+  gauge_param.ga_pad = 0;
 
 #ifdef MULTI_GPU
-  int x_face_size = gaugeParam->X[1]*gaugeParam->X[2]*gaugeParam->X[3]/2;
-  int y_face_size = gaugeParam->X[0]*gaugeParam->X[2]*gaugeParam->X[3]/2;
-  int z_face_size = gaugeParam->X[0]*gaugeParam->X[1]*gaugeParam->X[3]/2;
-  int t_face_size = gaugeParam->X[0]*gaugeParam->X[1]*gaugeParam->X[2]/2;
+  int x_face_size = gauge_param.X[1] * gauge_param.X[2] * gauge_param.X[3] / 2;
+  int y_face_size = gauge_param.X[0] * gauge_param.X[2] * gauge_param.X[3] / 2;
+  int z_face_size = gauge_param.X[0] * gauge_param.X[1] * gauge_param.X[3] / 2;
+  int t_face_size = gauge_param.X[0] * gauge_param.X[1] * gauge_param.X[2] / 2;
   int pad_size =MAX(x_face_size, y_face_size);
   pad_size = MAX(pad_size, z_face_size);
   pad_size = MAX(pad_size, t_face_size);
-  gaugeParam->ga_pad = pad_size;
+  gauge_param.ga_pad = pad_size;
 #endif
+}
 
-
+void setInvertParam(QudaInvertParam &inv_param)
+{
   // Solver params
-
-  inv_param->verbosity = QUDA_VERBOSE;
-  inv_param->mass = mass;
-  inv_param->kappa = kappa = 1.0/(8.0 + mass); // for Laplace operator
-  inv_param->laplace3D = laplace3D;            // for Laplace operator
+  inv_param.verbosity = QUDA_VERBOSE;
+  inv_param.mass = mass;
+  inv_param.kappa = kappa = 1.0 / (8.0 + mass); // for Laplace operator
+  inv_param.laplace3D = laplace3D;              // for Laplace operator
 
   // outer solver parameters
-  inv_param->inv_type = inv_type;
-  inv_param->tol = tol;
-  inv_param->tol_restart = 1e-3; // now theoretical background for this parameter...
-  inv_param->maxiter = niter;
-  inv_param->reliable_delta = reliable_delta;
-  inv_param->use_alternative_reliable = alternative_reliable;
-  inv_param->use_sloppy_partial_accumulator = false;
-  inv_param->solution_accumulator_pipeline = solution_accumulator_pipeline;
-  inv_param->pipeline = pipeline;
-
-  inv_param->Ls = Nsrc;
-
-  if(tol_hq == 0 && tol == 0){
+  inv_param.inv_type = QUDA_BICGSTAB_INVERTER; // Dummy setting
```

Comment:
@cpviolator Why did you make this change? The user should be able to specify whichever inverter they want.